client/opentracing/src/main/java/io/opentracing/APMSpan.java

```diff
@@ -59,22 +59,22 @@
      * @param builder  The span builder
      * @param recorder The trace recorder
      */
-    public APMSpan(APMSpanBuilder builder, TraceRecorder recorder, ContextSampler sampler) {
+    public APMSpan(APMSpanBuilder builder, TraceRecorder recorder, ContextSampler sampler, String serviceName, String buildStamp) {
```

Comment:
What I don't like about the Map approach is that it requires the consumer to know the map keys, whereas a dedicated constructor is a contract that is well-specified.